### PR TITLE
PMM-13082 Backwards compatibility with PMM v2

### PR DESCRIPTION
**Pull request by ademidoff**
PMM-13082

Link to the Feature Build: SUBMODULES-3730

### CODE REVIEW

File: managed/services/agents/grpc/agentpb_server.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+// agentPBServer provides methods for pmm-agent <-> pmm-managed interactions.
+type agentPBServer struct {
+	handler *agents.Handler
+
+	agentpb.UnimplementedAgentServer
+}
+
+// NewAgentServer creates new agent server.
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/services/agents/grpc/agentpb_server.go:31]: Fixed.

File: api/buf.yaml [hunk 2]
```diff
@@ -7,18 +7,28 @@ deps:
   - buf.build/grpc-ecosystem/grpc-gateway
 lint:
   use:
-    - DEFAULT
+    - STANDARD
```
Comment (ademidoff) [inline api/buf.yaml:10]: FYI: DEFAULT has become deprecated.

**Review by BupycHuk — APPROVED** [review]

File: managed/services/qan/client.go [hunk 3]
```diff
@@ -180,7 +182,8 @@ func (c *Client) Collect(ctx context.Context, metricsBuckets []*agentv1.MetricsB
 
 	convertedMetricsBuckets := make([]*qanv1.MetricsBucket, 0, len(metricsBuckets))
 	for _, m := range metricsBuckets {
-		agent := agents[m.Common.AgentId]
+		agentID, _ := strings.CutPrefix(m.Common.AgentId, "/agent_id/")
+		agent := agents[agentID]
```
Comment (BupycHuk) [inline managed/services/qan/client.go:186]: is it really a case?
it should be QAN Agent ID, not pmm-agent ID, so this value should come to pmm-agent from the server

Comment (ademidoff) [reply managed/services/qan/client.go:186]: Yeah, it believe it's legit. We'd better discuss.

**Review by JiriCtvrtka — APPROVED** [review]

**Pull request merged**